File: WeddingInfo.Frontend/src/app/models/Event.ts
import { IEvent } from './interfaces/IEvent';
import { IUser } from './interfaces/IUser';
import { EventType } from './enums/EventType';
import { EventDifficulty } from './enums/EventDifficulty';
import { User } from './User';
import { IMedia } from './interfaces/IMedia';

export class Event implements IEvent {
    public address: string;
    public description: string;
    public eventDifficulty: EventDifficulty;
    public eventType: EventType;
    public guests: IUser[];
    public id: number;
    public images: IMedia[];
    public isWeddingEvent: boolean;
    public isWeddingPartyEvent: boolean;
    public name: string;
    public occurance: Date;
    public order: number;
    public videos: IMedia[];

    public constructor(event?: IEvent) {
        this.address = event && event.address || '';
        this.description = event && event.description || '';
        this.eventDifficulty = event && event.eventDifficulty || EventDifficulty.Easy;
        this.eventType = event && event.eventType || EventType.Outdoors;
        this.guests = event && event.guests.map(g => new User(g)) || [];
        this.id = event && event.id || 0;
        this.images = event && event.images || [];
        this.isWeddingEvent = event && event.isWeddingEvent || false;
        this.isWeddingPartyEvent = event && event.isWeddingPartyEvent || false;
        this.name = event && event.name || '';
        this.occurance = event && event.occurance && new Date(event.occurance) || new Date();
        this.order = event && event.order || 0;
        this.videos = event && event.videos || [];
    }

}

File: WeddingInfo.Domain/Models/Event.cs
using System;
using System.Collections.Generic;
using WeddingInfo.Domain.Enumerations;

namespace WeddingInfo.Domain.Models
{
    public class Event
    {
		public int Id { get; set; }
		public string Name { get; set; }
		public string Description { get; set; }
        public int Order { get; set; }
		public DateTime Occurance { get; set; }
		public string Address { get; set; }
		public IList<User> Guests { get; set; } = new List<User>();
		public IList<Media> Images { get; set; } = new List<Media>();
        public IList<Media> Videos { get; set; } = new List<Media>();
		public EventType EventType { get; set; }
		public EventDifficulty EventDifficulty { get; set; }
		public bool IsWeddingEvent { get; set; }
		public bool IsWeddingPartyEvent { get; set; }
    }
}
